app/Community/Controllers/GameCommentController.php

```diff
@@ -4,23 +4,69 @@
 
 namespace App\Community\Controllers;
 
-use App\Community\Actions\AddCommentAction;
 use App\Community\Actions\GetUrlToCommentDestinationAction;
-use App\Community\Requests\CommentRequest;
-use App\Models\Comment;
+use App\Community\Data\CommentData;
+use App\Community\Data\GameCommentsPagePropsData;
+use App\Community\Data\SubscriptionData;
+use App\Community\Enums\SubscriptionSubjectType;
+use App\Community\Requests\StoreCommentRequest;
+use App\Data\PaginatedData;
 use App\Models\Game;
 use App\Models\GameComment;
+use App\Models\Subscription;
+use App\Models\User;
+use App\Platform\Data\GameData;
 use Illuminate\Contracts\View\View;
 use Illuminate\Http\RedirectResponse;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+use Inertia\Response as InertiaResponse;
 
 class GameCommentController extends CommentController
 {
-    public function index(Game $game): View
+    public function index(Game $game): InertiaResponse|RedirectResponse
     {
         $this->authorize('viewAny', [GameComment::class, $game]);
 
-        return view('game.comment.index')
-            ->with('game', $game);
+        $perPage = 50;
+        $currentPage = (int) request()->input('page', 1);
+
+        // Get total comments to calculate the last page.
+        $totalComments = $game->visibleComments()->count();
+        $lastPage = (int) ceil($totalComments / $perPage);
+
+        // If the current page exceeds the last page, redirect to the last page.
+        if ($currentPage !== 1 && $currentPage > $lastPage) {
+            return redirect()->route('game.comment.index', ['game' => $game->id, 'page' => $lastPage]);
+        }
+
+        $paginatedComments = $game->visibleComments()
+            ->with(['user' => function ($query) {
+                $query->withTrashed();
+            }])
+            ->paginate(50);
+
+        /** @var ?User $user */
+        $user = Auth::user();
+        $subscription = null;
+        if ($user) {
+            $subscription = Subscription::whereUserId($user->id)
+                ->whereSubjectType(SubscriptionSubjectType::GameWall)
+                ->whereSubjectId($game->id)
+                ->first();
+        }
+
+        $props = new GameCommentsPagePropsData(
+            game: GameData::fromGame($game)->include('badgeUrl', 'system'),
+            paginatedComments: PaginatedData::fromLengthAwarePaginator(
+                $paginatedComments,
+                total: $paginatedComments->total(),
+                items: CommentData::fromCollection($paginatedComments->getCollection())
+            ),
+            subscription: $subscription ? SubscriptionData::from($subscription) : null,
+        );
+
+        return Inertia::render('game/[game]/comments', $props);
```

Comment:
Not sure where the page title is coming from, but it lost a colon.

Before:
![image](https://github.com/user-attachments/assets/1711737d-8c29-429f-adff-e19fae22e21f)

After:
![image](https://github.com/user-attachments/assets/f3ae6f60-e6a4-408a-a76f-3b37f0063347)

But maybe it should be game title first (supported game files looks like it's not rendering correctly):
![image](https://github.com/user-attachments/assets/74c318bb-262f-497b-965a-48871751a0f3)

Reply:
Good catch on both issues. Fixed in latest.